prusti-specs/src/extern_spec_rewriter.rs

```diff
@@ -201,76 +136,470 @@ fn rewrite_method_inputs(item_ty: &syn::Type, method: &mut ImplItemMethod) ->
+            let mut struct_impl: syn::ItemImpl = parse_quote_spanned! {self.item_trait.span()=>
+                #[allow(non_camel_case_types)]
+                impl< #generic_params > #struct_ident < #generic_params > #where_clause {}
+            };
+
+            // Add items to impl block
+            for trait_item in self.item_trait.items.iter() {
+                match trait_item {
+                    syn::TraitItem::Type(_) => {
+                        // Ignore associated types, they are encoded as generics in the struct
+                    }
+                    syn::TraitItem::Method(trait_method) => {
+                        let method = self.generate_method_stub(trait_method);
+                        struct_impl.items.push(syn::ImplItem::Method(method));
+                    }
+                    _ => unimplemented!("Unimplemented trait item for extern spec"),
+                };
+            }
+
+            struct_impl
+        }
+
+        /// Generates a "stub" implementation for a trait method
+        fn generate_method_stub(&self, trait_method: &syn::TraitItemMethod) -> syn::ImplItemMethod {
+            let mut trait_method_sig = trait_method.sig.clone();
+
+            // Rewrite occurrences of associated types in signature to defined generics
+            syn::visit_mut::visit_signature_mut(
+                &mut AssociatedTypeRewriter::new(&self.assoc_types_to_generics_map),
+                &mut trait_method_sig,
+            );
+            let trait_method_ident = &trait_method_sig.ident;
+
+            // Rewrite "self" to "_self" in method attributes and method inputs
+            let mut trait_method_attrs = trait_method.attrs.clone();
+            trait_method_attrs
+                .iter_mut()
+                .for_each(|attr| attr.tokens = rewrite_self(attr.tokens.clone()));
+            let trait_method_inputs =
+                rewrite_method_inputs(&self.self_type_param_ident, &mut trait_method_sig.inputs);
+
+            // Create the method signature
+            let trait_ident = &self.item_trait.ident;
+            let method_path: syn::ExprPath = parse_quote_spanned! {trait_method_ident.span()=>
+                #trait_ident :: #trait_method_ident
+            };
+
+            // Create method
+            return parse_quote_spanned! {trait_method.span()=>
+                #[trusted]
+                #[prusti::extern_spec]
+                #(#trait_method_attrs)*
+                #[allow(unused)]
+                #trait_method_sig  {
+                    #method_path ( #trait_method_inputs );
+                    unimplemented!();
+                }
+            };
+        }
+    }
+
+    fn get_associated_types(item_trait: &syn::ItemTrait) -> Vec<&syn::TraitItemType> {
+        let mut result = Vec::new();
+        for trait_item in item_trait.items.iter() {
+            if let syn::TraitItem::Type(assoc_type) = trait_item {
+                // TODO: How to handle associated types with defaults?
```

Comment:
I think specifying a default for an associated type in an `extern_spec` block is probably a user error.